Gems/Atom/RHI/DX12/Code/Source/RHI/FrameGraphCompiler.cpp

```diff
@@ -413,13 +411,15 @@ namespace AZ
                 return;
             }
 
+            Scope& firstScope = static_cast<Scope&>(scopeAttachment->GetScope());
+            Buffer& buffer = static_cast<Buffer&>(*bufferFrameAttachment.GetBuffer()->GetDeviceBuffer(firstScope.GetDeviceIndex()));
```

Comment:
Using the device buffer of the device of the first scope won't work if the buffer is used on multiple devices. You need to submit barriers for the right device buffers. The same is true for the image barriers

Reply:
Thanks for pointing that out, we checked if that can be solved easily but tracking the pre-discards and all transitions actually requires more than we currently have at this stage in the code and is something that we would like to tackle through the FrameGraph properly, by linking Attachments on a per-device basis and tracking dependencies correctly.
In which case, the current code at this point would be valid again.
I noted it as a comment in the current version.